contracts/v3/strategies/MIMConvexStrategy.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import '../interfaces/IConvexVault.sol';
+import '../interfaces/IStableSwap2Pool.sol';
+import './BaseStrategy.sol';
+
+contract MIMConvexStrategy is BaseStrategy {
+    // used for Crv -> weth -> [mim/3crv] -> mimCrv route
+    address public immutable crv;
+    address public immutable cvx;
+
+    address public immutable mim;
+    address public immutable cvx3;
+
+    uint256 public immutable pid;
+    IConvexVault public immutable convexVault;
+    address public immutable mimCvxDepositLP;
+    IConvexRewards public immutable crvRewards;
+    IStableSwap2Pool public immutable stableSwap2Pool;
+
+    constructor(
+        string memory _name,
+        address _want,
+        address _crv,
+        address _cvx,
+        address _weth,
+        address _mim,
+        address _cvx3,
+        uint256 _pid,
+        IConvexVault _convexVault,
+        IStableSwap2Pool _stableSwap2Pool,
+        address _controller,
+        address _manager,
+        address _router
+    ) public BaseStrategy(_name, _controller, _manager, _want, _weth, _router) {
+        (, address _token, , address _crvRewards, , ) = _convexVault.poolInfo(_pid);
+        crv = _crv;
+        cvx = _cvx;
+        mim = _mim;
+        cvx3 = _cvx3;
+        pid = _pid;
+        convexVault = _convexVault;
+        mimCvxDepositLP = _token;
+        crvRewards = IConvexRewards(_crvRewards);
+        stableSwap2Pool = _stableSwap2Pool;
+        // Required to overcome "Stack Too Deep" error
+        _setApprovals(
+            _want,
+            _crv,
+            _cvx,
+            _mim,
+            _cvx3,
+            address(_convexVault),
+            address(_stableSwap2Pool)
+        );
+    }
+
+    function _setApprovals(
+        address _want,
+        address _crv,
+        address _cvx,
+        address _mim,
+        address _cvx3,
+        address _convexVault,
+        address _stableSwap2Pool
+    ) internal {
+        IERC20(_want).safeApprove(address(_convexVault), type(uint256).max);
+        IERC20(_crv).safeApprove(address(router), type(uint256).max);
+        IERC20(_cvx).safeApprove(address(router), type(uint256).max);
+        IERC20(_mim).safeApprove(address(_stableSwap2Pool), type(uint256).max);
+        IERC20(_cvx3).safeApprove(address(_stableSwap2Pool), type(uint256).max);
+        IERC20(_want).safeApprove(address(_stableSwap2Pool), type(uint256).max);
+    }
+
+    function _deposit() internal override {
+        convexVault.depositAll(pid, true);
```

Comment:
The implementation slightly differs from https://github.com/code-423n4/2021-09-yaxis/blob/main/contracts/v3/strategies/NativeStrategyCurve3Crv.sol#L56

The implementation in NativeStrategyCurve3Crv also does
```solidity
        uint256 _wantBal = balanceOfWant();
        if (_wantBal > 0) {
```

I know that this check is done before calling `_deposit`, but for the sake of consistency, I would say, keep both of them the same.